sbor/src/encode.rs

```diff
@@ -38,8 +38,19 @@ impl<'a, X: CustomTypeId> Encoder<'a, X> {
         self.write_slice(discriminator.as_bytes());
     }
 
-    pub fn write_size(&mut self, len: usize) {
-        self.buf.extend(&(len as u32).to_le_bytes());
+    pub fn write_size(&mut self, mut size: usize) {
+        // LEB128 and 4 bytes max
+        assert!(size <= 0x0FFFFFFF); // 268,435,455
```

Comment:
So if we're encoding bytes, this sets a max size of about 250MB. This feels... eminently reasonable? 👍 

That said, this should be an EncodeError, not an assert - I guess we can fix this in a future PR when we add the EncodeError?

Reply:
Yes, was planning to do that in one batch.